### ISSUE TITLE
Update MCMC sampler wrappers in Turing.jl to use AbstractMCMC

### ISSUE THREAD

**Issue by mhauru** [TuringLang/Turing.jl#2426]
We currently have lots of glue code bridging the gap between sampler packages like AdvancedMH and AdvancedHMC and Turing's implementation of `sample`. We should be able to remove the vast majority now using the AbstractMCMC interface.

This needs to be done after #2318, which will release us from using all the space `Selector` stuff, which makes up a lot of the glue code.

## Related issues
* https://github.com/TuringLang/Turing.jl/issues/2367

**Comment by torfjelde** [comment]
Is this not a duplicate of #2413 ?

**Comment by mhauru** [comment]
Seems to be
